pkg/apis/coordination/types.go

```diff
@@ -67,3 +93,69 @@ type LeaseList struct {
 	// items is a list of schema objects.
 	Items []Lease
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// LeaseCandidate defines a candidate for a lease object.
+// Candidates are created such that coordinated leader election will pick the best leader from the list of candidates.
+type LeaseCandidate struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+	Spec LeaseCandidateSpec
+}
+
+// LeaseCandidateSpec is a specification of a Lease.
+type LeaseCandidateSpec struct {
+	// LeaseName is the name of the lease for which this candidate is contending.
+	// This field is immutable.
+	LeaseName string
+	// PingTime is the last time that the server has requested the LeaseCandidate
+	// to renew. It is only done during leader election to check if any
+	// LeaseCandidates have become ineligible. When PingTime is updated, the
+	// LeaseCandidate will respond by updating RenewTime.
+	// +optional
+	PingTime *metav1.MicroTime
+	// RenewTime is the time that the LeaseCandidate was last updated.
+	// Candidates are expected to update their renewTime periodically,
+	// but not as aggressively as if they were the leader.
+	// Any time a Lease needs to do leader election, the Ping field
+	// is updated to signal to the LeaseCandidate that they should update
+	// the RenewTime.
+	// Old LeaseCandidate objects are also garbage collected if it has been hours since the last renew.
+	// +optional
+	RenewTime *metav1.MicroTime
+	// BinaryVersion is the binary version. It must be in a semver format.
+	// This field is effectively required when Strategy is "OldestEmulationVersion"
+	// +optional
+	BinaryVersion string
+	// EmulationVersion is the emulation version. It must be in a semver format.
+	// This field is effectively required when Strategy is "OldestEmulationVersion"
+	// +optional
+	EmulationVersion string
+	// PreferredStrategies indicates the list of strategies for picking the leader for coordinated leader election.
+	// The list is ordered, and the first strategy supersedes all other strategies. The list is used by coordinated
+	// leader election to make a decision about the final election strategy. This follows as
+	// - If all clients have strategy X as the first element in this list, strategy X will be used.
+	// - If a candidate has strategy [X] and another candidate has strategy [Y, X], Y supersedes X and strategy Y
```

Comment:
```suggestion
	// - If a candidate has strategy [X ...] and another candidate has strategy [Y, X ...], Y supersedes X and strategy Y
```

Or do we say anything about the length? Why would we not want to restrict this field to be either one or two elements?

Reply:
Discussed offline, we'll support lengths longer than 2.